pilot/pkg/model/authentication_test.go

```diff
@@ -0,0 +1,247 @@
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+			},
+		},
+		{
+			name:              "Empty workload labels in baz",
+			workloadNamespace: "baz",
+			workloadLabels:    labels.Collection{},
+			want: []*Config{
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "istio-config",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+			},
+		},
+		{
+			name:              "Match workload labels in foo",
+			workloadNamespace: "foo",
+			workloadLabels:    labels.Collection{{"app": "httpbin", "version": "v1", "other": "lables"}},
+			want: []*Config{
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "foo",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "with-selector",
+						Namespace: "foo",
+					},
+					Spec: &security_beta.RequestAuthentication{
+						Selector: &selectorpb.WorkloadSelector{
+							MatchLabels: map[string]string{
+								"app":     "httpbin",
+								"version": "v1",
+							},
+						},
+					},
+				},
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "istio-config",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+			},
+		},
+		{
+			name:              "Match workload labels in bar",
+			workloadNamespace: "bar",
+			workloadLabels:    labels.Collection{{"app": "httpbin", "version": "v1"}},
+			want: []*Config{
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "bar",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "istio-config",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+			},
+		},
+		{
+			name:              "Paritial match workload labels in foo",
+			workloadNamespace: "foo",
+			workloadLabels:    labels.Collection{{"app": "httpbin"}},
+			want: []*Config{
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "foo",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+				&Config{
+					ConfigMeta: ConfigMeta{
+						Type:      "request-authentication",
+						Name:      "default",
+						Namespace: "istio-config",
+					},
+					Spec: &security_beta.RequestAuthentication{},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if got := policies.GetJwtPoliciesForWorkload(tc.workloadNamespace, tc.workloadLabels); !reflect.DeepEqual(tc.want, got) {
```

Comment:
`reflect.DeepEqual`, does the impl itself requires a specific list ordering? or does not matter we simply use the given impl's output for the test comparision?

Reply:
Config should be sorted by `[sortConfigByCreationTime](https://github.com/istio/istio/blob/d6827a9cabcefcc0d172d117e5b6079a157d4bed/pilot/pkg/model/config.go#L360)`, which also use name + namespace as the second tier. Thus, order should be deterministic